framework/helpers/BaseJson.php

```diff
@@ -179,4 +180,54 @@ protected static function processData($data, &$expressions, $expPrefix)
 
         return $data;
     }
+
+    /**
+     * Generates a summary of the validation errors.
+     * @param Model|Model[] $models the model(s) whose validation errors are to be displayed.
+     * @param array $options the tag options in terms of name-value pairs. The following options are specially handled:
+     *
+     * - showAllErrors: boolean, if set to true every error message for each attribute will be shown otherwise
+     *   only the first error message for each attribute will be shown. Defaults to `false`.
+     *
+     * @return string the generated error summary
+     * @since 2.0.14
+     */
+    public static function errorSummary($models, $options = [])
+    {
+        $showAllErrors = ArrayHelper::remove($options, 'showAllErrors', false);
+        $lines = self::collectErrors($models, $showAllErrors);
+
+        return json_encode($lines);
+    }
+
+    /**
+     * Return array of the validation errors
+     * @param Model|Model[] $models the model(s) whose validation errors are to be displayed.
+     * @param $showAllErrors boolean, if set to true every error message for each attribute will be shown otherwise
+     * only the first error message for each attribute will be shown.
+     * @return array of the validation errors
+     * @since 2.0.14
+     */
+    private static function collectErrors($models, $showAllErrors)
```

Comment:
this realisation is equal to BaseConsole::collecErrors. Time for generalization is came 😸 .

Reply:
@dmirogin yeah, it's debatable thing: https://github.com/yiisoft/yii2/pull/15268#pullrequestreview-80677027

Reply:
i don't like this way. Better to create new class for that.

Reply:
@dmirogin what class name/placement do you suggest?

Reply:
something like this
~~~php
<?php

class ErrorsCollector
{
    /**
     * @var array Model[]
     */
    private $models;
    /**
     * @var bool
     */
    private $showAllErrors;

    public function __construct($models, $showAllErrors = false)
    {
        $this->models = $this->ensureModelsIsArray($models);
        $this->showAllErrors = $showAllErrors;
    }

    /**
     * @return array
     */
    public function getLines()
    {
        $lines = [];
        
        foreach ($this->getModelsErrors() as $error) {
            if (!in_array($error, $lines, true)) {
                $lines[] = $error;
            }
            if (!$this->showAllErrors) {
                break;
            }
        }
        
        return $lines;
    }

    /**
     * @return array
     */
    public function getModelsErrors()
    {
        $modelErrors = [];
        foreach ($this->models as $model) {
            /* @var $model Model */
            foreach ($model->getErrors() as $errors) {
                $modelErrors[] = $errors;
            }
        }
        
        return $modelErrors;
    }

    /**
     * @param Model[]|Model $models
     * @return array
     */
    private function ensureModelsIsArray($models)
    {
        if (!is_array($models)) {
            return [$models];
        }

        return $models;
    }
}

~~~

Reply:
Makes sense.

Reply:
@dmirogin, I disagree that we need another separate class that will work with Model errors. It should be Model responsibility.

Reply:
Ok, now we have 3 identical methods in 3 different classes. Maybe worth consider trait for that ?

Reply:
As you see, codeclimate ok with this 2 identical (they are similar but not the same (c) samdark) methods in 2 different classes, because these methods now are just a couple of lines. And overall in the code base, there are a hundred of places with such "identical" methods. 
@MysteryDragon, @saidbakr  need your opinion as well.

Reply:
In this particular case these methods aren't same as I said before. I don't think it worth extracting two foreach-es in a separate class.

Reply:
Well, it is OK either way...